scripts/o3de/o3de/android_support.py

```diff
@@ -0,0 +1,2116 @@
+                               tool_version_arg='--version',
+                               version_regex=r'.*(cmake version)\s*\"?([\d\_\.]+)',
+                               android_config=android_config)
+
+
+def validate_ninja(android_config) -> (Path, str):
+    """
+    Specialization of validate_build_tool for Ninja Build
+    :param android_config:      The android configuration to get the tool home if needed
+    :return:    Tuple of : The full path of the tool and the tool version
+    """
+    return validate_build_tool(tool_name='Ninja',
+                               tool_command=f'ninja{EXE_EXTENSION}',
+                               tool_config_key=None,
+                               tool_environment_var=None,
+                               tool_config_sub_path='',
+                               tool_version_arg='--version',
+                               version_regex=r'(\s*)([\d\_\.]+)',
+                               android_config=android_config)
+
+
+PLATFORM_SETTINGS_FORMAT = """
+# Auto Generated from last cmake project generation ({generation_timestamp})
+
+[settings]
+platform={platform}
+game_projects={project_path}
+asset_deploy_mode={asset_mode}
+
+[android]
+android_sdk_path={android_sdk_path}
+android_gradle_plugin={android_gradle_plugin_version}
+"""
+
+PROJECT_DEPENDENCIES_VALUE_FORMAT = """
+dependencies {{
+{dependencies}
+    api 'androidx.core:core:1.1.0'
+}}
+"""
+
+NATIVE_CMAKE_SECTION_ANDROID_FORMAT = """
+    externalNativeBuild {{
+        cmake {{
+            buildStagingDirectory "{native_build_path}"
+            version "{cmake_version}"
+            path "{absolute_cmakelist_path}"
+        }}
+    }}
+"""
+
+NATIVE_CMAKE_SECTION_DEFAULT_CONFIG_NDK_FORMAT_STR = """
+        ndk {{
+            abiFilters '{abi}'
+        }}
+"""
+
+OVERRIDE_JAVA_SOURCESET_STR = """
+            java {{
+                srcDirs = ['{absolute_azandroid_path}', 'src/main/java']
+            }}
+"""
+
+NATIVE_CMAKE_SECTION_BUILD_TYPE_CONFIG_FORMAT_STR = """
+            externalNativeBuild {{
+                cmake {{
+                    {targets_section}
+                    arguments {arguments}
+                }}
+            }}
+"""
+
+CUSTOM_APPLY_ASSET_LAYOUT_TASK_FORMAT_STR = """
+    task syncLYLayoutMode{config}(type:Exec) {{
+        workingDir '{working_dir}'
+        commandLine {full_command_line}
+    }}
+
+    process{config}MainManifest.dependsOn syncLYLayoutMode{config}
+
+    syncLYLayoutMode{config}.mustRunAfter {{
+        tasks.findAll {{ task->task.name.contains('strip{config}DebugSymbols') }}
+    }}
+
+"""
+
+CUSTOM_GRADLE_COPY_REGISTRY_FOLDER_FORMAT_STR = """
+     task copyRegistryFolder{config}(type: Copy) {{
+        from ('build/intermediates/cmake/{config_lower}/obj/arm64-v8a/{config_lower}/Registry')
+        into ('{asset_layout_folder}/registry')
+        include ('*.setreg')
+    }}
+
+    merge{config}Assets.dependsOn copyRegistryFolder{config}
+"""
```

Comment:
I am assuming this is a copy and paste from the old android_support.py.
Wouldn't this not work as the Registry folder wouldn't be found?\

There is an easier solution to this problem though and that is to just generate the `cmake_dependencies.*.setreg` file outside of the build folder.

Actually one easy solution is that we can generate the cmake_dependencies into a `~/.o3de/ActiveGems/<project-name>/Registry` or hardlink it there as well.
Such as in [SettingsRegistry.cmake](https://github.com/o3de/o3de/blob/development/cmake/SettingsRegistry.cmake#L309) after generating the cmake_dependencies.*.setreg we can run the [file(CREATE_LINK)](https://cmake.org/cmake/help/latest/command/file.html#create-link) command.

```cmake
o3de_get_home_path(home_path)
cmake_path(APPEND home_path "ActiveGems" OUTPUT_VARIABLE mirrored_dependencies_setreg)
if (prefix)
    cmake_path(APPEND mirrored_dependencies_setreg "${prefix}")
else()
    cmake_path(APPEND mirrored_dependencies_setreg "__NO_PROJECT__")
endif()
cmake_path(APPEND mirrored_dependencies_setreg "Registry/cmake_dependencies.${specialization_name}.setreg")
file(CREATE_LINK "${dependencies_setreg}" "${mirrored_dependencies_setreg}")
```

Reply:
You are right, this was copy pasted and shouldn't be here. The task isn't running anyways, and the source of the copy isn't valid anymore. I will remove this whole section